pkg/util/server/error_test.go

```diff
@@ -54,12 +54,10 @@ func Test_writeError(t *testing.T) {
 		t.Run(tt.name, func(t *testing.T) {
 			rec := httptest.NewRecorder()
 			WriteError(tt.err, rec)
-			require.Equal(t, tt.expectedStatus, rec.Result().StatusCode)
-			b, err := ioutil.ReadAll(rec.Result().Body)
-			if err != nil {
-				t.Fatal(err)
-			}
-			require.Equal(t, tt.msg, string(b[:len(b)-1]))
+			res := &ErrorResponseBody{}
+			json.NewDecoder(rec.Result().Body).Decode(res)
+			require.Equal(t, tt.expectedStatus, res.Code)
```

Comment:
Hmmm shouldn't you match `tt.expectedStatus` with `res.Status` instead?

Reply:
`res.Status` is a string, indicating `success` or `error`, whereas `Code` is the actual status code of the response.
However, `rec.Result().StatusCode` should be equal to `res.Code` as well.